torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -221,6 +221,52 @@ const std::vector<std::string> functions = {
 
             # FIXME: torchscript: torch.zeros(sizes, grad.options())
             return torch.zeros(sizes).to(grad).scatter_(dim, indices, grad)
+        
+        # python implementation of at::_gather_sparse_backward function. Need to Refine
+        def AD_gather_sparse_backend(grad,
+                                        self,
+                                        dim: int,
+                                        index):
+            if (self.dim() == 0) :
+                zero_grad = torch.zeros(0, grad.numel()).to(index)
+                return torch._sparse_coo_tensor_unsafe(zero_grad, grad, self.size())
+            if (grad.dim() == 0) :
+                return torch._sparse_coo_tensor_unsafe(index.view(1, 1), grad, self.size())
+
+            grad_self = torch.zeros_like(self)
+            # FIXME: torch.empty(self.dim(), grad.numel(), self.options()).dtype(torch.long)
+            sparse_ind = torch.empty(self.dim(), grad.numel()).to(index)
+            print(sparse_ind)
+
+            grad_numel = grad.numel()
+            if (grad_numel > 0) :
+                n_above = grad_numel
+                n_below = 1
+                if (dim < 0):
+                    dim += self.dim()
+                for i in range(self.dim()) :
+                    n_above = int(n_above / grad.size(i))
+                    if (i == dim) :
+                        sparse_ind[i] = index.reshape(-1)
+                    else: 
+                        sparse_ind[i] = torch.arange(grad.size(i)).unsqueeze(1).expand(grad.size(i), n_above).reshape(-1).repeat(n_below).to(index)
+                    
+                    n_below *= grad.size(i)
+            return torch._sparse_coo_tensor_unsafe(sparse_ind, grad.reshape(-1), self.size())
+
```

Comment:
`AD_gather_sparse_backend` is just a python implementation of `at::_gather_sparse_backward`, which is the grad function of `torch.gather` when `sparse_grad` is `True`.